agent_verification/src/generic_oid4vc.rs

```diff
@@ -0,0 +1,92 @@
+use oid4vc_core::authorization_request::{Body, Object};
+use oid4vp::oid4vp::OID4VP;
+use serde::{Deserialize, Serialize};
+use siopv2::siopv2::SIOPv2;
+
+// TODO(oid4vc): All types and functionalities in this file should be implemented properly in the `oid4vc` crates.
+
+pub type SIOPv2AuthorizationResponse = oid4vc_core::authorization_response::AuthorizationResponse<SIOPv2>;
+pub type OID4VPAuthorizationResponse = oid4vc_core::authorization_response::AuthorizationResponse<OID4VP>;
+pub type SIOPv2AuthorizationRequest = oid4vc_core::authorization_request::AuthorizationRequest<Object<SIOPv2>>;
+pub type OID4VPAuthorizationRequest = oid4vc_core::authorization_request::AuthorizationRequest<Object<OID4VP>>;
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+#[serde(untagged)]
+pub enum GenericAuthorizationResponse {
```

Comment:
Maybe a little docs (`///`) would help to understand why we need a "generic" authorization response.